// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.confidential_match.lookup_server.proto_backend;

import "protos/lookup_server/backend/location.proto";
import "protos/lookup_server/backend/sharding_scheme.proto";

option java_package = "com.google.cm.lookupserver.backend";
option java_outer_classname = "DataExportInfoProto";

// Represents a data export.
message DataExportInfo {
  // A unique ID tied to the data export.
  string data_export_id = 1;

  // Location of the object storage directory containing the shard.
  Location shard_location = 2;

  // Location of the object storage directory containing the metadata.
  Location metadata_location = 3;

  // The scheme used to shard the data.
  ShardingScheme sharding_scheme = 4;
}
